src/System.ServiceProcess.ServiceController/tests/System.ServiceProcess.ServiceController.Tests/ServiceControllerTests.cs

```diff
@@ -218,43 +220,77 @@ public void GetDevices()
+                {
+                    AssertHasDependent(dependent, _testService.DependentTestServiceNamePrefix + j, _testService.DependentTestServiceDisplayNamePrefix + j);
+                }
+
+                // Assert that the dependent service depends on the main test service
+                AssertDependsOn(dependent, _testService.TestServiceName, _testService.TestServiceDisplayName);
+
+                // Assert that this dependent service depends on all the test services created before it
+                Assert.Equal(i + 1, dependent.ServicesDependedOn.Length);
+
+                for (int j = i - 1; j >= 0; j--)
+                {
+                    AssertDependsOn(dependent, _testService.DependentTestServiceNamePrefix + j, _testService.DependentTestServiceDisplayNamePrefix + j);
+                }
             }
         }
 
-        [Fact]
-        public void ServicesDependedOn()
+        public void Dispose()
         {
-            // The test service creates a number of dependent services, each of these should depend on the test service
-            var controller = new ServiceController(_testService.TestServiceName);
-            Assert.True(controller.DependentServices.Length > 0);
+            _testService.DeleteTestServices();
+        }
 
-            for (int i = 0; i < controller.DependentServices.Length; i++)
+        private static ServiceController AssertHasDependent(ServiceController controller, string serviceName, string displayName)
+        {
+            var dependent = FindService(controller.DependentServices, serviceName, displayName);
+            if (dependent == null)
             {
-                var dependent = controller.DependentServices[i];
-                Assert.True(dependent.ServicesDependedOn.Length == 1);
+                Assert.True(false, string.Format("Expected service {0} to have dependent service {1}", controller.ServiceName, serviceName));
+            }
+
+            return dependent;
+        }
 
-                var dependency = dependent.ServicesDependedOn[0];
-                Assert.Equal(_testService.TestServiceName, dependency.ServiceName);
-                Assert.Equal(_testService.TestServiceDisplayName, dependency.DisplayName);
+        private static ServiceController AssertDependsOn(ServiceController controller, string serviceName, string displayName)
+        {
+            var dependency = FindService(controller.ServicesDependedOn, serviceName, displayName);
+            if (dependency == null)
+            {
+                Assert.True(false, string.Format("Expected service {0} to depend on service {1}", controller.ServiceName, serviceName));
             }
+
+            return dependency;
         }
 
-        public void Dispose()
+        private static ServiceController FindService(ServiceController[] services, string serviceName, string displayName)
         {
-            _testService.DeleteTestServices();
+            foreach (ServiceController service in services)
```

Comment:
Here I avoided taking the dependency on an additional NuGet package just for the sake of using IEnumerable<T>.Where() once.  If there were more code where LINQ applied, I would take the dependency.